_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,656 @@
+
+To take the LN case of HTLCs as an example, we can produce two
+independent scripts, each of which handles one of the first two items
+above:
+
+    (1) OP_HASH256 <hash> OP_EQUALVERIFY <Alice pubkey> OP_CHECKSIG
+    (2) <time delta> OP_CHECKSEQUENCEVERIFY OP_DROP <Bob pubkey> OP_CHECKSIG
+
+These separate scripts are then hashed so that they can be used as the
+leaves of a merkle tree.  As described earlier, the data to be hashed is
+prefixed with a tag (itself hashed) indicating its purpose.  "TapLeaf"
+is the tag in this case.  The leaves must also include the size of the
+script and a version (only version 0xc0 is defined in this proposal;
+other versions are available for future soft fork upgrades).
+
+After the two scripts are hashed, their digests are put in lexicographic
+order.  This ordering will allow later verification of a merkle
+inclusion proof without knowing whether or not each leaf or branch
+appeared in the tree to the left or right of its pair sibling, thus
+reducing the amount of data that needs to be communicated as well as
+potentially improving privacy.  After ordering, the two hashes will be
+hashed together with the prefix tag "TapBranch".  As this merkle tree
+only has two leaves, the resultant hash is the merkle root.
+
+    FIXME: diagram of tree
+
+This merkle tree only covers two of the HTLC's possible end states.  For
+the third case where Alice and Bob mutually agree on a spend, they can
+use signature aggregation using something like MuSig.  As in the
+multisig section above, they work together to create a single pubkey,
+called the *Taproot internal key.*
+
+The merkle root and the internal key are then hashed together (prefix
+tag, "TapTweak") and the resultant digest is used as a private key from
+which a pubkey is derived, this value being known as the *tweak.*  The
+tweak pubkey is added to the internal key (the same operation used in
+[BIP32][] HD key derivation) in order to derive the *taproot output
+key---*the key that is used in any bech32 addresses and scriptPubKeys
+that pay these three conditions.
+
+    FIXME: diagram of tweaking, similar to one from BitMex article
+
+When it comes time to spend this money, either Alice or Bob can provide
+the script they want to use, the data needed by it (a signature and,
+sometimes, a hash preimage), the Taproot internal key, and the hash of
+the script they didn't use.  Alternatively, Alice and Bob can work together
+to use signature aggregation (after accounting for the tweak) to provide
+a signature in combination with the same single-key, single-signature
+spending pattern described in the previous subsections.  As long as the
+data they provide in either case is correct, the spend will be accepted.
+
+{% comment %}<!--
+    P2WSH
+      scriptPubKey: 34
+      witnessScript: OP_IF OP_HASH256 OP_EQUALVERIFY <A pubkey> OP_CHECKSIG
+                     OP_ELSE <delta> OP_CSV OP_DROP <B pubkey> OP_CHECKSIG OP_ENDIF
```

Comment:
The `<B` is parsed as HTML (bold) which messes up the rendered version. Perhaps `&lt;B` can be used instead.

Reply:
@practicalswift I suggest you look at the version rendered by Netlify rather than GitHub, since all the links and images and other site-specific stuff will work there.  (In the GitHub status box near the bottom of this page, expand the "All checks has passed" item and click the *Details* link to the right of "Deploy preview is ready".)